scientific_library/tvb/simulator/simulator.py

```diff
@@ -278,12 +278,13 @@ def configure(self, full_configure=True):
 
         """
         # TODO: temporary hack because numba dfuns fail for multiple modes
-        if self.use_numba and self.model.number_of_modes == 1:
-            self._dfun = self.model.dfun
-        else:
+        self._dfun = self.model.dfun
+        if not self.use_numba or self.model.number_of_modes > 1:
             self.use_numba = False
-            self._dfun = self.model._numpy_dfun
+            if hasattr(self.model, "_numpy_dfun"):
+                self._dfun = self.model._numpy_dfun
```

Comment:
This type of code is ugly, especially as we are having both `Model` and `ModelNumbaDfun` base classes very easy to use/switch.

Reply:
I would agree type dispatch is better, though numba gufuncs support modes, it's really the dfun method wrapping it which assumes otherwise.  You rather need to think about which stage you're coupling the modes.